ee/katc/config.go

```diff
@@ -98,36 +100,43 @@ func (r *rowTransformStep) UnmarshalJSON(data []byte) error {
 	}
 }
 
-// katcTableConfig is the configuration for a specific KATC table. The control server
-// sends down these configurations.
-type katcTableConfig struct {
-	SourceType        katcSourceType     `json:"source_type"`
-	Source            string             `json:"source"` // Describes how to connect to source (e.g. path to db) -- % and _ wildcards supported
-	Platform          string             `json:"platform"`
-	Columns           []string           `json:"columns"`
-	Query             string             `json:"query"` // Query to run against `path`
-	RowTransformSteps []rowTransformStep `json:"row_transform_steps"`
-}
+type (
+	// katcTableConfig is the configuration for a specific KATC table. The control server
+	// sends down these configurations.
+	katcTableConfig struct {
+		Columns           []string                 `json:"columns"`
+		SourceType        katcSourceType           `json:"source_type"`
+		SourcePaths       []string                 `json:"source_paths"` // Describes how to connect to source (e.g. path to db) -- % and _ wildcards supported
+		SourceQuery       string                   `json:"source_query"` // Query to run against each source path
+		RowTransformSteps []rowTransformStep       `json:"row_transform_steps"`
+		Overlays          []katcTableConfigOverlay `json:"overlays"`
```

Comment:
I suspect this should be `[]katcTableConfig`

Reply:
Discussed in huddle, adding extra context here for others reviewing --

I had originally settled on keeping the structs `katcTableConfig` and `katcTableConfigOverlay` separate because I didn't think it made sense to allow `SourceType`, `SourcePaths`, `SourceQuery`, and `RowTransformSteps` to be nil at the top level (the `katcTableConfig` level), but I felt that it was necessary to allow them to be nil at the overlay level (the `katcTableConfigOverlay` struct).

I am considering:

```
type (
	// katcTableConfig is the configuration for a specific KATC table. The control server
	// sends down these configurations.
	katcTableConfig struct {
		Columns []string `json:"columns"`
		katcTableDefinition
		Overlays []katcTableConfigOverlay `json:"overlays"`
	}

	katcTableConfigOverlay struct {
		Filters map[string]string `json:"filters"` // determines if this overlay is applicable to this launcher installation
		katcTableDefinition
	}

	katcTableDefinition struct {
		SourceType        *katcSourceType     `json:"source_type,omitempty"`
		SourcePaths       *[]string           `json:"source_paths,omitempty"` // Describes how to connect to source (e.g. path to db) -- % and _ wildcards supported
		SourceQuery       *string             `json:"source_query,omitempty"` // Query to run against each source path
		RowTransformSteps *[]rowTransformStep `json:"row_transform_steps,omitempty"`
	}
)
```

However, I'm still not sure I like allowing `SourceType` to be nil at the top level. I am going to mull it over some more and come back tomorrow! But wanted to leave my thoughts in case folks wanted to chime in.

Reply:
Adding -- I did not want to combine the structs completely because I think it is incorrect to allow an overlay to define `Columns`.

Reply:
> However, I'm still not sure I like allowing SourceType to be nil at the top level. I am going to mull it over some more and come back tomorrow! But wanted to leave my thoughts in case folks wanted to chime in.

Why not?

> I did not want to combine the structs completely because I think it is incorrect to allow an overlay to define Columns.

Why not?

Reply:
re SourceType:

Feels messy? I should just try implementing it and see if it actually feels annoying to work around or not.

re columns:

I think the columns that a table returns should be consistent.

Reply:
I would expect us to need to vary source by platform _somewhere_. That's exactly the kind of nonsense I expect. We see this commonly in things like registry values vs plist files, but it's possible that it will make more sense to handle _that_ skew in SQL and not here.

Reply:
columns I could go either way for. I agree it feels a little weird to vary. Kinda touches your question about dataflatten. I think we should do whatever for columns, and we can update it if we find use cases.

Reply:
Ok, I'll go with the option I posted above, which will let source type be nil at the top level!